src/OData.Cli/Program.cs

```diff
@@ -1,18 +1,77 @@
-﻿using Microsoft.Extensions.CommandLineUtils;
+﻿using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.Threading.Tasks;
 
 namespace OData.Cli
 {
     class Program
     {
-        static void Main(string[] args)
+        static async Task Main(string[] args)
         {
-            var app = new CommandLineApplication();
+            var app = new RootCommand
+            {
+                new Command("generate", "Command to generate proxy classes for OData endpoints.")
+                {
+                    new Option<string>(
+                        new[] { "--metadata", "-m" },
+                        "The URI of the metadata document. The value must be set to a valid service document URI or a local file path."
+                    ){IsRequired = true},
+                    new Option<string>(
+                        new[] { "--custom-headers|--h"},
+                        "Headers that will get sent along with the request when fetching the metadata document from the service."
+                    ),
+                    new Option<string>(
+                        new[] { "--proxy|--p"},
+                        "Proxy settings. Format: domain\\user:password@SERVER:PORT."
+                    ),
+                    new Option<string>(
+                        new[] { "--namespace|--ns"},
+                        "The namespace of the client code generated."
+                    ),
+                    new Option<bool?>(
+                        new[] { "--no-tracking|-n"},
+                        "The use of this disables entity and property tracking."
```

Comment:
```suggestion
                        "Disable entity and property tracking."
```
I think you should add more info to explain what the impact of disabling entity/property tracking is.